### PR TITLE
Fix resume track button

### PR DESCRIPTION
**Pull request by andy1li**
**Checklist**:

- [x] I've thoroughly self-reviewed my changes
- [x] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [x] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **New Features**
	- Enhanced routing functionality for the Resume Track button, allowing the inclusion of query parameters when transitioning to the course route.
	- Added visibility checks and interaction capabilities for the "resume track" button based on user authentication and course progress.

- **Bug Fixes**
	- Improved handling of course tracking by ensuring the correct parameters are passed during navigation.

- **Tests**
	- Introduced new acceptance tests for the track page functionality, verifying the behavior of the "resume track" button for different user states and course progress.
	- Updated test organization for better clarity and structure in acceptance tests related to the track page.
	- Enhanced page object functionality with new properties for interacting with and checking the visibility of the "resume track" button.

### CODE REVIEW

**Comment by rohitpaulk** [comment]
@andy1li let's add a test for this? This is a pretty common user flow, so I think it's worth adding a test to cover this.

File: app/components/track-page/resume-track-button.ts [hunk 1]
```diff
@@ -27,16 +27,17 @@ export default class CourseOverviewResumeTrackButtonComponent extends Component<
     }
 
     return this.args.courses.find((course) => {
-      return !this.authenticator
-        .currentUser!.repositories.filterBy('course', course)
-        .filterBy('language', this.args.language)
-        .some((repository) => repository.allStagesAreComplete);
+      const courseRepositories = this.authenticator.currentUser!.repositories.filterBy('course', course).filterBy('language', this.args.language);
+
+      return courseRepositories.length > 0 && !courseRepositories.some((repository) => repository.allStagesAreComplete);
```
Comment (andy1li) [inline app/components/track-page/resume-track-button.ts:32]: There's a subtle bug for course with zero repos:
- `![].some` is true, 
- but we don't want to resume a course with zero repos.

File: tests/pages/track-page.js [hunk 2]
```diff
@@ -12,6 +12,9 @@ export default createPage({
     await finishRender(); // Page has poller
   },
 
+  clickResumeTrackButton: clickable('[data-test-resume-track-button]'),
```
Comment (rohitpaulk) [inline tests/pages/track-page.js:15]:
```suggestion
  clickOnResumeTrackButton: clickable('[data-test-resume-track-button]'),
```

(consistent)

File: app/components/track-page/resume-track-button.ts (hunk 1 above)
Comment (rohitpaulk) [reply app/components/track-page/resume-track-button.ts:32]: @andy1li Can you clarify this? `[].some()` shouldn't return true if there are no elements (as far as I know).

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Question added

File: app/components/track-page/resume-track-button.ts (hunk 1 above)
Comment (rohitpaulk) [reply app/components/track-page/resume-track-button.ts:32]: Oh... k got it, `some` returns false, but there are zero repos. Cool!

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**